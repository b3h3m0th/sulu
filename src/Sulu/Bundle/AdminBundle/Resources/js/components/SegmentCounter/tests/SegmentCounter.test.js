// @flow
import React from 'react';
import {render} from '@testing-library/react';
import SegmentCounter from '../SegmentCounter';

jest.mock('../../../utils/Translator', () => ({
    translate: jest.fn((key) => key),
}));

test('Should show a positive count if nothing is passed', () => {
    const {container} = render(<SegmentCounter delimiter="," max={20} value={undefined} />);
    expect(container).toMatchSnapshot();
});

test('Should show a positive count', () => {
    const {container} = render(<SegmentCounter delimiter="," max={5} value="keyword1, keyword2, keyword3" />);
    expect(container).toMatchSnapshot();
});

test('Should show a negative count', () => {
    const {container} = render(<SegmentCounter delimiter="|" max={2} value="That|is|a|test" />);
    expect(container).toMatchSnapshot();
});
