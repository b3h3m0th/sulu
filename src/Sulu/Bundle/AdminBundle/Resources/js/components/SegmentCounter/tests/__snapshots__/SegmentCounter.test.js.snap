// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should show a negative count 1`] = `
<div>
  <label
    class="segmentCounter exceeded"
  >
    -2
     
    sulu_admin.segments_left
  </label>
</div>
`;

exports[`Should show a positive count 1`] = `
<div>
  <label
    class="segmentCounter"
  >
    2
     
    sulu_admin.segments_left
  </label>
</div>
`;

exports[`Should show a positive count if nothing is passed 1`] = `
<div>
  <label
    class="segmentCounter"
  >
    20
     
    sulu_admin.segments_left
  </label>
</div>
`;
